napari/_qt/qt_viewer.py

```diff
@@ -330,6 +353,8 @@ def _add_layer(self, layer):
         layer : napari.layers.Layer
             Layer to be added.
         """
+        layer.events.select.connect(self._on_active_layer_change)
+        layer.events.deselect.connect(self._on_active_layer_change)
```

Comment:
When thinking about the most common scenario function `_on_active_layer_change` is called twice for each change of the active layer (one fore selection, and one for deselection). I do not like this.

Reply:
I'd actually like the drop the `deselect` method, but even then there will be a trigger from each layer that changes. So if you have 5 selected layers and then click to unselect all 5 and a 6th gets selected you will have 6 calls. It's not clear to me how to change that (currently happens without this PR too)

Reply:
yes please!  I'd like to see a `layer.events.selected` where `event.value` has the current state of selection.

Reply:
> yes please! I'd like to see a layer.events.selected where event.value has the current state of selection.

I can do that in it's own PR, which will then make this one simpler. Will get started on that soon